tensorflow_addons/layers/normalizations.py

```diff
@@ -321,3 +321,208 @@ def __init__(self, **kwargs):
+
+    Input shape
+        Arbitrary. Use the keyword argument `input_shape`
+        (tuple of integers, does not include the samples axis)
+        when using this layer as the first layer in a model. This layer, as of now,
+        works on a 4-D tensor where the tensor should have the shape [N X H X W X C]
+
+        TODO: Add support for more input shapes, especially for FC layers.
+
+    Output shape
+        Same shape as input.
+
+    References
+        - [Filter Response Normalization Layer: Eliminating Batch Dependence
+        in the training of Deep Neural Networks]
+        (https://arxiv.org/abs/1911.09737)
+    """
+
+    def __init__(
+        self,
+        epsilon: float = 1e-6,
+        axis: list = [1, 2],
+        beta_initializer: types.Initializer = "zeros",
+        gamma_initializer: types.Initializer = "ones",
+        beta_regularizer: types.Regularizer = None,
+        gamma_regularizer: types.Regularizer = None,
+        beta_constraint: types.Constraint = None,
+        gamma_constraint: types.Constraint = None,
+        learned_epsilon: bool = False,
+        name: str = None,
+        **kwargs
+    ):
+        super().__init__(name=name, **kwargs)
+        self.epsilon = tf.cast(epsilon, dtype=self.dtype)
+        self.beta_initializer = tf.keras.initializers.get(beta_initializer)
+        self.gamma_initializer = tf.keras.initializers.get(gamma_initializer)
+        self.beta_regularizer = tf.keras.regularizers.get(beta_regularizer)
+        self.gamma_regularizer = tf.keras.regularizers.get(gamma_regularizer)
+        self.beta_constraint = tf.keras.constraints.get(beta_constraint)
+        self.gamma_constraint = tf.keras.constraints.get(gamma_constraint)
+        self.use_eps_learned = learned_epsilon
+        self.supports_masking = True
+
+        if self.use_eps_learned:
+            self.eps_learned_initializer = tf.keras.initializers.Constant(1e-4)
+            self.eps_learned = self.add_weight(
+                shape=(1,),
+                name="learned_epsilon",
+                dtype=self.dtype,
+                initializer=tf.keras.initializers.get(self.eps_learned_initializer),
+                regularizer=None,
+                constraint=tf.keras.constraints.non_neg,
+            )
+        else:
+            self.eps_learned_initializer = None
+
+        self._check_axis(axis)
+
+    def build(self, input_shape):
+        if len(tf.TensorShape(input_shape)) != 4:
+            raise ValueError(
+                """Only 4-D tensors (CNNs) are supported
+        as of now."""
+            )
+        self._check_if_input_shape_is_none(input_shape)
+        self._create_input_spec(input_shape)
+        self._add_gamma_weight(input_shape)
+        self._add_beta_weight(input_shape)
+        super().build(input_shape)
+
+    def call(self, inputs):
+        epsilon = tf.math.abs(self.epsilon)
+        if self.use_eps_learned:
+            epsilon += self.eps_learned
```

Comment:
This is incorrect. The paper mentions that `self.eps_learned` should be `abs`. We can add a check in `__init__` to ensure that `epsilon` passed by the user is also positive. 
```suggestion
        epsilon = self.epsilon
        if self.use_eps_learned:
            epsilon += tf.math.abs(self.eps_learned)
```

Reply:
There is a `constraint` associated with this parameter that makes sure that it is always non-negative. Isn't that enough?

Reply:
My bad didn't see the `non_neg` constraint. You can ignore the `abs` around `self.eps_learned`. 

Weirdly enough, `non_neg` constraint doesn't use `abs`, it does [this](https://github.com/tensorflow/tensorflow/blob/5943793102a96fb47e5bc007576e1f7f41600bb6/tensorflow/python/keras/constraints.py#L94) instead. No idea why.

Reply:
Actually that is correct. `non-neg` mathematically means a lower bound of zero. There should be an `abs` for epsilon instead of this constraint.